### DISCUSSION TITLE
Not able to generate APK for testing.

### DISCUSSION THREAD

**Discussion by harmit001** [OneKeyHQ/app-monorepo#2253] [Q&A] [answered]
I was trying to generate the APK for testing but not able to generate it. I have a review the guideline but not able to find the steps.

**Comment by workspaceking** [comment]
I am facing the same issue, did you figure it out!!!!

**Reply by assetgrowth80-arch** [reply]
What the issue you are facing

**Comment by rayston92** [comment]
@sidmorizon @sunnylqm

**Comment by sunnylqm** [comment]
Please provide the errors you met

**Reply by workspaceking** [reply]
it throws webpack version error

**Reply by sunnylqm** [reply]
Please use expo-cli@6.1.0 , or any version below 6.1.0

**Comment by LiWiTf** [comment]
I want to build my own apk but can't find a way

**Comment by revan-zhang** [comment, marked as answer]
Please using the new v5.x version and try again. We’ve updated a lot of infrastructure information.

**Comment by assetgrowth80-arch** [comment]
Hi, I understand what you’re trying to accomplish. You’re essentially looking for a watch-only Bitcoin payment setup where your website can generate receiving addresses from the public key of your OneKey Pro, while the private keys remain safely offline on the hardware wallet.

In principle, the OneKey Pro does not need to stay connected or powered on just to receive Bitcoin. The important part is that the addresses generated by your code must use the exact same Bitcoin account, network, derivation path, script/address type, and address-index scheme used by the OneKey wallet. Otherwise, you can generate a valid BTC address that isn’t recognized as belonging to your OneKey account.

Regarding the $5 test transaction, don’t assume that it is lost. If the transaction was confirmed, the BTC is almost certainly still sitting at the address you sent it to. The first thing to do is identify the exact address you generated and check it on a Bitcoin block explorer. Then compare that address with the receiving addresses generated by the OneKey account.

I would also avoid using exported logs or any private/secret material for this. You only need the appropriate extended public key (xpub/ypub/zpub, depending on the account type) for address derivation. The private key/seed should remain on the OneKey device.

Since OneKey support specifically directed you to GitHub, this is a good issue to raise with them. Include the wallet/account type, Bitcoin network, the public-key type you exported, derivation path, address type you generated, and the test address/transaction ID. Do not post your seed phrase or private key.

OneKey’s official developer portal has documentation for hardware integration, and the official GitHub repositories include the app and hardware SDK, so the developers should be able to confirm the exact derivation parameters for the OneKey Pro. 

If you provide the public address you generated, the transaction ID, and the exact public-key type/path you used (with all private information removed), the derivation problem can be diagnosed without exposing your wallet’s private keys.

OneKey official GitHub repository⁠￼
OneKey Developer Portal⁠￼

One caution: I would not tell this person that the $5 is definitely recoverable until the destination address and transaction are checked. If the address was derived from the wrong account/key, the funds may still be recoverable if the corresponding private key is actually controlled by the user, but that needs to be established first.
